core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -129,16 +141,29 @@ jagged_vector_buffer<TYPE>::jagged_vector_buffer(
         ((host_access_resource != nullptr) ? m_outer_memory.get()
                                            : m_outer_host_memory.get());
 
+    // Size of the data payload.
+    const std::size_t dataSize =
+        std::accumulate(capacities.begin(), capacities.end(),
+                        static_cast<std::size_t>(0)) *
+        sizeof(TYPE);
+
     // Set up the host accessible memory array.
-    std::ptrdiff_t ptrdiff =
+    void* unaligned_start_ptr =
+        m_inner_memory.get() +
         (capacities.size() * sizeof(typename value_type::size_type));
+    std::size_t buffer_size =
+        m_inner_memory_size -
+        (capacities.size() * sizeof(typename value_type::size_type));
+    char* start_ptr = static_cast<char*>(
+        std::align(alignof(TYPE), dataSize, unaligned_start_ptr, buffer_size));
+    std::ptrdiff_t ptrdiff = 0;
```

Comment:
Call me stupid, but this code could use some more comments explaining what's happening.

Reply:
Really, it's not obvious right away? :smile: After I composed the lines with having 4-5 different tabs open that I would switch between, to try to come up with the correct incantation... :stuck_out_tongue: